### ISSUE TITLE
Certain ConfigContexts cause NullPointerExceptions despite having documented defaults

### ISSUE THREAD

**Issue by tjcelaya** [joyent/java-manta#247]
**TL;DR: we aren't handling some configuration parameters in the execution path which have documented default values**

Despite the defaults listed in [README.md](README.md#configuration), certain implementations of `ConfigContext` result in `NullPointerException`s. Adding getters for `disableNativeSignatures` and `noAuth` to these classes would save some trouble for developers that choose to use these classes. One option is to implement getters in these classes if the behavior of null is best decided by the class itself. Most of these classes use `MantaUtils.parseBooleanOrNull`

~Problematic Classes~ Classes that do what they say they do:
 - `MapConfigContext`: uses `MantaUtils.parseBooleanOrNull` which returns null when passed null
 - `EnvVarConfigContext`: uses `MantaUtils.parseBooleanOrNull` as well
 - `StandardConfigContext`: doesn't implement anything, which is probably as intended and implies the null-handling for default behavior belongs outside these classes
 - ~`ChainedConfigContext`~: abstracts over arrays of multiple `ConfigContext`s and probably shouldn't be concerned with this kind of thing

Alternatively, we could modify the lines that cause the `NullPointerException`s codepath to check handle nulls explicitly, specifically the following two lines
 - https://github.com/joyent/java-manta/blob/master/java-manta-client/src/main/java/com/joyent/manta/client/MantaClient.java#L207
 - https://github.com/joyent/java-manta/blob/master/java-manta-client/src/main/java/com/joyent/manta/http/MantaConnectionFactory.java#L171 Interestingly enough line 165 in that same file does explicit null checking for `useNativeCodeToSign`

Code used to verify which classes throw and which don't:

https://gist.github.com/tjcelaya/df63bfe932afff07de4f501d6daff241

**Comment by dekobon** [comment]
The fundamental problem is that we don't have a stable value for **UNSET**. I could see us refactor to a wrapping object that would allow for that information to be stored, but that would be a fair bit of work.
